cmd/kubeadm/app/cmd/init.go

```diff
@@ -345,3 +351,25 @@ func (i *Init) Run(out io.Writer) error {
 
 	return initDoneTempl.Execute(out, ctx)
 }
+
+func createClientsetAndOptionallyWaitForReady(cfg *kubeadmapi.MasterConfiguration, dryRun bool) (clientset.Interface, error) {
+	if dryRun {
+		// If we're dry-running; we should create a faked client that answers some GETs in order to be able to do the full init flow and just logs the rest of requests
+		dryRunGetter := apiclient.NewInitDryRunGetter(cfg.NodeName, cfg.Networking.ServiceSubnet)
```

Comment:
So there are a number of different clientset stubs, I'm wondering if we can dedupe.